hazelcast/src/main/java/com/hazelcast/internal/metrics/managementcenter/MetricsResultSet.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright (c) 2008-2019, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.internal.metrics.managementcenter;
+
+import javax.annotation.Nonnull;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import static com.hazelcast.internal.metrics.managementcenter.MetricsCompressor.decompressingIterator;
+
+public class MetricsResultSet {
+
+    private final long nextSequence;
+    private final List<MetricsCollection> collections;
+
+    public MetricsResultSet(long nextSequence, List<Map.Entry<java.lang.Long, byte[]>> collections) {
+        //        this.nextSequence = slice.nextSequence();
+        this.nextSequence = nextSequence;
+        this.collections = collections.stream()
```

Comment:
A minor improvement can be removing this convertion. 
We can just keep `List<Map.Entry<java.lang.Long, byte[]>>`  instead of a `List<MetricsCollection>`
MetricCollection just provide an iterator which delegates to `decompressingIterator(bytes)`
MC codebase can directly call this static method.

Reply:
Removed the conversion